src/reducers/book.js

```diff
@@ -4,13 +4,13 @@ const books = (state = [], action) => {
   switch (action.type) {
     case CREATE_BOOK:
       return [
-        ...state,
         {
           ...action.book,
         },
+        ...state,
       ];
     case REMOVE_BOOK:
-      return state; // to be modified in milestone 4
+      return state.filter(book => book.id !== action.book.id);
```

Comment:
We assume `book.id` is unique here but it's not hence the bug shown below:

#### The `gif` below shows how the remove button does not work for duplicate ids.
![bookerrbugvictor](https://user-images.githubusercontent.com/36057474/73345457-683a7800-4284-11ea-9e69-23660d28ab37.gif)


I would recommend using a more unique generator, see https://gist.github.com/6174/6062387

Reply:
Thank you @bolah2009 for the review.

I knew beforehand it will get to this point when I saw that the Pathright suggested using Math.random for id generation. I am sure this will not be the case if the app used a better id generator or getting data from a back-end that ensures uniqueness. 

In actual implementation, IDs are unique, and that should take care of the bug.

Thanks for sharing that link :+1:  I will use it to simulate unique ID for the app.

Reply:
:thinking: @bolah2009, I would like to know how you made this ```gif```. It looks really nice and simple

Reply:
> The random guid generator you shared will suffice for now.

Yes, it's using the same `Math.random` but I believe generates more unique values.

> I would like to know how you made this gif. It looks really nice and simple

Oh, thanks!

I am using **Peek**, it's on Unbuntu Store - http://peek.uploadedlobster.com/